src/charm.py

```diff
@@ -399,6 +400,15 @@ def _update_hydra_endpoints_relation_data(self, event: RelationEvent) -> None:
 
     def _on_hydra_pebble_ready(self, event: WorkloadEvent) -> None:
         """Event Handler for pebble ready event."""
+        # Necessary directory for log forwarding
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to Hydra container. Deferring event.")
+            self.unit.status = WaitingStatus("Waiting to connect to Hydra container")
+            return
+        if not self._container.isdir(self._log_dir):
+            self._container.make_dir(path=self._log_dir, make_parents=True, permissions=0o777)
```

Comment:
Shouldn't the permissions be a little more restricted?

Reply:
changed to 755